cudamapper/include/claragenomics/cudamapper/overlapper.hpp

```diff
@@ -59,6 +64,33 @@ class Overlapper
     /// \brief Identified overlaps which can be combined into a larger overlap and add them to the input vector
     /// \param overlaps reference to vector of Overlaps. New overlaps (result of fusing) are added to this vector
     static void post_process_overlaps(std::vector<Overlap>& overlaps);
+
+    /// \brief Extends a single overlap at its ends if the similarity of the query and target sequences is above a specified threshold.
+    /// \param overlap An Overlap which is modified in place. Any of the query_start_position_in_read, query_end_position_in_read,
+    /// target_start_position_in_read, and target_end_position_in_read fields may be modified.
+    /// \param query_sequence A std::string_view of the query read sequence.
+    /// \param target_sequence A std::string_view of the target read sequence.
+    /// \param extension The number of bases to extend at the head and tail of the overlap. If the head or tail is shorter than extension,
+    /// the function only tries to extend to the end of the read.
+    /// \param required_similarity The minimum similarity to require to extend an overlap.
+    static void extend_overlap_by_sequence_similarity(Overlap& overlap,
```

Comment:
`rescue_overlap_ends()` calls `extend_overlap_by_sequence_similarity()`. We don't call `extend_overlap...` externally as things stand but it seems like a useful function to have around. Tests operate on `extend_overlap_by_sequence_similarity()` as well and I assume this means it needs to be exposed.

Reply:
It that case I would suggest doing something similar to this: https://github.com/clara-parabricks/ClaraGenomicsAnalysis/blob/e43a18c4288f57ee554f81391905a24e159dda44/cudamapper/src/index_gpu.cuh#L158

In other words creating
```
namespace details
{
namespace overlapper
{
    void extend_overlap_by_sequence_similarity(){ ... }
}
}
```

That way the function can still be tested independently (which is a problem with protected/private), but it is clear that the function is not meant for external use (this can be changed if such need arises)

Reply:
Awesome - fixed in c4d7caa.